examples/wave/wave-op-mpi.py

```diff
@@ -141,15 +141,20 @@ def bump(actx, dcoll, t=0):
             / source_width**2))
 
 
-def main(ctx_factory, dim=2, order=3, visualize=False):
+def main(ctx_factory, dim=2, order=3, visualize=False, lazy=False):
     cl_ctx = ctx_factory()
     queue = cl.CommandQueue(cl_ctx)
-    actx = PyOpenCLArrayContext(
+    actx_outer = PyOpenCLArrayContext(
         queue,
         allocator=cl_tools.MemoryPool(cl_tools.ImmediateAllocator(queue)),
         force_device_scalars=True,
     )
 
+    if lazy:
+        actx_rhs = PytatoPyOpenCLArrayContext(queue)
+    else:
+        actx_rhs = actx_outer
```

Comment:
Rather than have two different `actx` floating around, why not just strictly use the pytato context throughout the script when in lazy-mode? I would just kill `actx_outer` entirely. It should work just fine, as I do that in my ESDG examples (which strictly use lazy): https://github.com/inducer/grudge/blob/ESDG-development/examples/euler/acoustic_pulse.py#L236

Reply:
I agree with that. While it's nice to be able to use two array contexts, it's kind of clunky.